File: src/main/java/com/booknav/springbackend/dto/CategoryDTO.java
package com.booknav.springbackend.dto;

import com.booknav.springbackend.dto.Book.BookDTO;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class CategoryDTO {
    private Integer categoryId;
    @NotBlank(message = "Category title is required")
    @Size(max = 100, message = "Category title can't exceed 100 characters.")
    private String title;
    private List<BookDTO> books;
}

File: ASP Core - Backend/Dtos/CategoryDTO.cs
using System.ComponentModel.DataAnnotations;

namespace aspcorebackend.Dtos {
    public class CategoryDTO {
        public int CategoryId { get; set; }
        [MaxLength(150, ErrorMessage = "Category title can't exceed 150 characters.")]
        [Required(ErrorMessage = "Category title is required")]
        public string Title { get; set; } = null!;
        public DateTime AddedAt { get; set; }
        public List<BookDTO> Books { get; set; } = [];

        public CategoryDTO() {
            AddedAt = DateTime.Now;
        }
    }
}
